src/lazyload-image.ts

```diff
@@ -30,11 +30,27 @@ export function isVisible(element: HTMLElement, threshold = 0, _window = window)
     );
 }
 
-function loadImage(imagePath: string): Observable<HTMLImageElement> {
+function loadImage(element: HTMLElement, imagePath: string, useSrcset: boolean, sourcesAttrName: string): Observable<HTMLImageElement> {
```

Comment:
I agree, I didn't like passing around sourcesAttrName either.
I had no idea about higher order functions.

This is a great idea, and I've committed most of the changes.

I did not add the clonePicture helper function, since I'm not a huge fan of one-liner functions that are only used in one place.
It would be better if the function also retrieved and returned the child image element, like this:
```TypeScript
export function cloneImageWithPicture(element: HTMLImageElement): HTMLImageElement {
    const parentClone = element.parentNode.cloneNode(true) as HTMLPictureElement;
    return parentClone.getElementsByTagName('img')[0];
}
```
But I have no idea how to call a function where you pass an element, whose parent element has to be cloned and its child element has to be returned. So I just left it as it is.

I also did some name changes:
`isChildToPicture` => `isChildOfPicture`
Using **Of** here sound more natural. While I'm not a native English speaking person, from the little research I did - **To** is used in a more metaphorical sense, while **Of** is used in a more literal sense.

`setLazySource` => `setSourcesToLazy`
I think it might be a little unclear to what `Source` does the function refer to. And I don't think a function, that processes *possibly* multiple source elements, should be named in a singular form.
My suggestion is not perfect either, I'd much rather prefer `setSourcesToLazyImage` or `setSourcesToLazyImg`, but the name might be getting too long.